src/Server.php

```diff
@@ -11,15 +11,15 @@ class Server extends EventEmitter implements ServerInterface
 {
     private $io;
 
-    public function __construct(SocketServerInterface $io)
+    public function __construct(SocketServerInterface $io, RequestParserInterface $parser = null)
     {
         $this->io = $io;
 
-        $this->io->on('connection', function (ConnectionInterface $conn) {
+        $this->io->on('connection', function (ConnectionInterface $conn) use ($parser) {
             // TODO: http 1.1 keep-alive
             // TODO: chunked transfer encoding (also for outgoing data)
 
-            $parser = new RequestParser();
+            $parser = $parser ?: new RequestParser();
```

Comment:
I'm all for (somehow) injecting the parser instance to use here :+1: 

However, unfortunately the parser is stateful, meaning a single instance can't really be reused for multiple connections.